packages/vitest/src/types/config.ts

```diff
@@ -175,14 +178,14 @@ export interface InlineConfig {
   testNamePattern?: string | RegExp
 
   /**
-   * Listen to port and serve API
+   * Serve API options.
    *
-   * When set to true, the default port is 55555
+   * When set to true, the default port is 51204.
    *
    * @internal WIP
    * @default false
    */
-  api?: boolean | number
+  api?: boolean | ApiConfig
```

Comment:
I think we should keep supporting `api` as a number for setting the port here. And also offer the same in the CLI. So `--api <port>` is a shortcut for `--api --port <port>` that will be the most common server option to change.

Reply:
We can do that with yargs?

Reply:
I mean that you just ignore `--port` and others if `--api` isn't true in the CLI

Reply:
but then we need another type, since we are passing the `UserOptions` from the `cli`, change `resolveConfig` to also provide `ApiConfig` from the `cli`?